### PR TITLE
feat(ESSNTL-3701): Implement resource-types endpoints

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-3701](https://issues.redhat.com/browse/ESSNTL-3701).
It implements the GET `/resource-types` and `/resource-types/inventory-groups` endpoints. The output is structured as described in the ADR [here](https://docs.google.com/document/d/1_zWWRpunieVbvuR6SMjP2fReA1YOS1XruLQaTnYjb6A/edit#). 

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/resource_type.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from app.serialization import serialize_group
+from lib.feature_flags import FLAG_INVENTORY_GROUPS
+from lib.feature_flags import get_flag_value
+from lib.middleware import rbac
+
+logger = get_logger(__name__)
+
+
+@api_operation
+@rbac(Permission.ADMIN)
```
[the remark this answers is not in the document]
Comment (kruai) [reply api/resource_type.py:26]: Good question. That might be something we need to address with [ESSNTL-4505](https://issues.redhat.com/browse/ESSNTL-4505), as we don't have enums beyond Host-specific stuff yet.

**Review by jpramos123 — DISMISSED** [review]
LGTM

**Review by fellipeh — APPROVED** [review]
LGTM

File: swagger/api.spec.yaml [hunk 2]
```diff
@@ -533,6 +533,53 @@ paths:
         '404':
           description: Group not found.
 
+  /resource-types:
```
Comment (thearifismail) [inline swagger/api.spec.yaml:536]: Can RBAC team be asked to change the wording in URL.  This whole exercise is about RBAC roles but the URLs say "resource_types". It is going to be confusing for customers.  "role_types" sound more appropriate.

**Review by thearifismail — COMMENTED** [review]
LGTM

**Review by thearifismail — APPROVED** [review]

**Comment by kruai** [comment]
Reverted back to the version that was already approved. PR checks are failing on all PRs, but this will not affect any existing endpoints, so I'm going to merge.

**Pull request merged**